cruise-control-metrics-reporter/src/test/java/com/linkedin/kafka/cruisecontrol/metricsreporter/utils/CCEmbeddedBroker.java

```diff
@@ -33,7 +31,7 @@ public CCEmbeddedBroker(Map<Object, Object> config) {
       // Also validates the config
       KafkaConfig kafkaConfig = KafkaConfig.apply(config);
       parseConfigs(config);
-      _kafkaServer = new KafkaServer(kafkaConfig, Time.SYSTEM, Option.empty(), new ArrayBuffer<>());
+      _kafkaServer = new KafkaServer(kafkaConfig, Time.SYSTEM, Option.empty(), false);
```

Comment:
This change is not compatible with Kafka `2.5`, `2.6`, or `2.7` -- i.e. this constructor change was introduced in `2.8` via https://github.com/apache/kafka/pull/9883/files#diff-3638ff970bc6766f9e570f76a6440966930b990af33e53e5b3db9d65f90264e9L124.

To add `2.8` support to this branch, we should be able to support a drop in replacement of Kafka dependencies between `2.5`, `2.6`, `2.7`, and `2.8`  in this branch without code changes.

Would it be possible to use reflection to check whether the expected constructor exists (i.e. pre- and post-2.8) and switch to the existing constructor here?

Reply:
So I see what you are saying but, this is just the client library used to talk to the Kafka cluster and Kafka 2.8 is able to talk to 2.5+. Or are you saying that `migrate_to_2_5` means "can be built/run with kafka 2.5+ on the classpath"?

Reply:
I am happy to do that work, I just wanted to confirm.

Reply:
> So I see what you are saying but, this is just the client library used to talk to the Kafka cluster and Kafka 2.8 is able to talk to 2.5+. Or are you saying that `migrate_to_2_5` means "can be built/run with kafka 2.5+ on the classpath"?

Correct :) We would like to ensure that this branch can be built and run with Kafka 2.5+ on the classpath, because some CC users, including LinkedIn, may choose to inject/use clients either from a different Kafka version or a different flavor of Kafka (e.g. [LinkedIn Kafka](https://github.com/linkedin/kafka)) that might be compatible with Kafka 2.5 - 2.8.